GetStream/stream-chat-swift
# [CIS-1427] Fix quoted messages not updated after edit

### 🔗 Issue Link
CIS-1427

### 🎯 Goal
Fix quoted messages not getting updated if that quoted message was updated.

### 🛠 Implementation
Whenever a cell is updated on the message list view, we check if the updated message is currently from a quoted message in the visible cells, if it is we updated those cells.

We only need to handle the visible cells since when scrolling the message list, the cells that are not visible will be updated automatically.

### 🎨 Changes

https://user-images.githubusercontent.com/12814114/146598964-1e841626-40ec-4cf2-90f6-9c49c517e1d0.mp4

### 🧪 Testing
- Create a message A
- Create message B quoting message A
- Edit Message A
- Quoted message A in message B should be updated

### ☑️ Checklist
